src/main/java/IO/DbReader.java

```diff
@@ -0,0 +1,53 @@
+
+package IO;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+
+import data.Database;
+
+/* DbReader class accesses blob information to read its contents. */
+public class DbReader {
+
+    private Database database;
+    private String runId;
+    private String type;
+    private InputStream in;
+
+    /* Comments out extension variable bc may not need it
+     private static final String extension = ".bin"; . */
+    private String fileName;
+    
+    /* Makes a reader object. */
+    public DbReader(Database database, String runId, String type) {
+        this.database = database;
+        this.runId = runId;
+        this.type = type;
+        fileName = database.findName(this.runId, this.type);
+    }
+
+    /*
+     * Return a stream object for reading from the file, eventually from the
+     * database, right now reading a file that includes the database name.
+     */
+    public InputStream read() {
+        in = database.readData(fileName);
+        return in;
+    }
+
+    /*
+     * Handle all actions that are necessary when done using the DbReader
+     * Object.
+     */
+    public void finish() {
+        try {
+            in.close();
```

Comment:
What if the user close the stream themselves?

Reply:
I'm thinking of adding to the comment that when using this API, user must use the finish() to close instead of closing stream directly, incase things need to be wrapped up when the user is done using the reader class this could all happen in finish